src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/PackageRestore/PackageRestoreUnconfiguredInputDataSource.cs

```diff
@@ -44,7 +44,7 @@ protected override IDisposable LinkExternalInput(ITargetBlock<RestoreInfo> targe
                 _project.Services,
                 projects => projects.Select(project => project.Services.ExportProvider.GetExportedValueOrDefault<IPackageRestoreConfiguredInputDataSource>())
                                     .Where(dataSource => dataSource != null)    // Filter out those without PackageReference
-                                    .Select(DropConfiguredProjectVersions),
+                                    .Select(DropConfiguredProjectVersions!),
```

Comment:
Why?

Reply:
The `Where` doesn't convert from `T?` to `T` unfortunately.

Updated in 360f28cb8 to use a new `WhereNotNull` extension method that propagates this information.